src/electrum.rs

```diff
@@ -653,7 +664,10 @@ impl Call {
                     .downcast_ref::<bitcoincore_rpc::Error>()
                     .and_then(extract_bitcoind_error)
                 {
-                    Some(e) => error_msg(&self.id, RpcError::DaemonError(e.clone())),
+                    Some(e) => match self.params.parse_rpc_error_code(e.code) {
+                        Some(err) => error_msg(&self.id, err),
```

Comment:
IIUC this is sending the underlying error message to the caller. And while it's not recommended to connect untrusted devices to electrs some people may do that and it'd be better to just log the message and send back "internal server error" to avoid leaking information.

Reply:
> IIUC this is sending the underlying error message to the caller.

It is not. I added `self.params.parse_rpc_error_code` to allow certain errors to be intercepted and replaced but the underlying error message is not sent to the caller